rom/dev/src/flow/warm_reset.rs

```diff
@@ -27,8 +28,15 @@ impl WarmResetFlow {
     /// * `env` - ROM Environment
     #[inline(never)]
     #[cfg_attr(not(feature = "no-cfi"), cfi_impl_fn)]
-    pub fn run(_env: &mut RomEnv) -> CaliptraResult<Option<FirmwareHandoffTable>> {
+    pub fn run(env: &mut RomEnv) -> CaliptraResult<Option<FirmwareHandoffTable>> {
         cprintln!("[warm-reset] ++");
+
+        // Check if previous Cold-Reset was successful.
+        if env.data_vault.rom_cold_boot_status() != ColdResetComplete.into() {
+            cprintln!("[warm-reset] Previous Cold-Reset was not successful.");
+            return Err(CaliptraError::ROM_WARM_RESET_UNSUCCESSFUL_PREVIOUS_COLD_RESET);
```

Comment:
wouldn't it be better to instead treat this case as another cold reset? Could we initiate a cold reset from here?

Reply:
Currently, that is not possible. All ColdReset datavault entries that were locked will remain locked on a warm-reset, and ROM won't be able to record cold-boot info on this path.

Reply:
What I meant here was to initiate a cold reset at the hardware level, not to try to run the cold boot flow while hardware is doing a warm reset.

Reply:
It's not safe to initiate a cold reset of Caliptra without also resetting the rest of the SoC; the measurements made early in boot would be lost and Caliptra would think that whatever SoC firmware is loading measurements into it is the SoC boot ROM, when in fact it could be attacker-controlled mutable firmware.

Reply:
This would require a hardware change. We can discuss this as a separate issue.